### PR TITLE
fix: clear reactions when receiving message edits (AR-2881) (AR-2849)

### PR DESCRIPTION
**Pull request by vitorhugods**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [X] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [X] contains a reference JIRA issue number like `SQPIT-764`
  - [X] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [X] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

1. When a message is edited, we should clear the current reactions it has. At the moment we’re not clearing it.
2. When a message with reactions is edited, further edits and reactions will not be handled. 

### Causes (Optional)

1. Was not implemented before.
2. The Reactions table will cause a Foreign Key Constraint Failure when we attempt to update the ID of a message during and `UPDATE`, and it was a silent error because we don't have this whole operation inside a transaction.

### Solutions

- Move the whole edit a text message to a single DB transaction
This had the bonus that we were exposing 3 DB operations just to edit a single text message, and these operations were not being used elsewhere (update ID and mark as edited). So these functions are now removed.

- Make sure we clear reactions before editing the message ID

#### Test Coverage

- [X] I have added automated test to this contribution

### Attachments

In the video below, I'm sending messages, reacting and editing from the Web wrapper, everything seems to be alright:

https://user-images.githubusercontent.com/9389043/209229742-1b11aa3b-bf86-4ded-a1ec-610edda0d952.mp4

----
#### PR Post Merge Checklist for internal contributors

 - [X] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: persistence/src/commonTest/kotlin/com/wire/kalium/persistence/dao/message/MessageTextEditTest.kt [hunk 1]
```diff
@@ -0,0 +1,138 @@
+package com.wire.kalium.persistence.dao.message
```
Comment (yamilmedina) [inline persistence/src/commonTest/kotlin/com/wire/kalium/persistence/dao/message/MessageTextEditTest.kt:1]: 👏

**Review by yamilmedina — CHANGES_REQUESTED** [review]
Just added some concerns about logging ids, looking flawless nonetheless 👏

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/receiver/conversation/message/ApplicationMessageHandler.kt [hunk 2]
```diff
@@ -187,7 +192,11 @@ internal class ApplicationMessageHandlerImpl(
                 )
             }
 
-            is MessageContent.TextEdited -> editTextHandler.handle(signaling, content)
+            is MessageContent.TextEdited -> {
+                kaliumLogger.withFeatureId(KaliumLogger.Companion.ApplicationFlow.MESSAGES)
+                    .w("EDITING MESSAGE: ORIGINAL ID: ${content.editMessageId}, NEW ID: ${signaling.id}")
```
[the remark this answers is not in the document]
Comment (vitorhugods) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/receiver/conversation/message/ApplicationMessageHandler.kt:197]: I really trusted that `QualifiedId.toString()` would obfuscate it already 😄

Fixing it.

**Review by yamilmedina — APPROVED** [review]
Boa 🪁

**Pull request merged**